Elka1907 bjs2 86056

elka1907 opened the pull request:
Двойной кэш-кэш

elka1907 committed:
Remove bjs2_85711 folder from main branch

elka1907 committed:
Remove bjsc_85655 folder from main branch

elka1907 committed:
Remove reverse folder from main branch

elka1907 committed:
Add complete Double Cache-Cache

SnurnicynKonstantin commented on src/main/java/school/faang/bjs2_86056/StudentDatabase.java:
```diff
@@ -0,0 +1,96 @@
+
+    public void addStudentToSubject(Student student, Subject subject) {
+        subjectStudents.putIfAbsent(subject, new ArrayList<>());
+        if (!subjectStudents.get(subject).contains(student)) {
```
А попробуй цик заменить одной операцией - `subjectStudents.get(subject).addAll(students);`
Получится так?

elka1907 replied:
мы же здесь добавляем одного студента, поэтому не подойдёт

SnurnicynKonstantin replied:
Да, сори

SnurnicynKonstantin commented on src/main/java/school/faang/bjs2_86056/StudentDatabase.java:
```diff
@@ -0,0 +1,96 @@
+
+    public List<Student> getStudentsForSubject(Subject subject) {
+        if (subject == null) {
+            return new ArrayList<>();
```
Вместо `new ArrayList<>();` разработчики любят использовать `Collections.emptyList()`.
Вообще в Collections много всего полезного. Так же и Set-ы можно создавать пустые или с значениями

elka1907 replied:
done

SnurnicynKonstantin commented on src/main/java/school/faang/bjs2_86056/StudentDatabase.java:
```diff
@@ -0,0 +1,96 @@
+        if (subject == null) {
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(subjectStudents.getOrDefault(subject, new ArrayList<>()));
```
А зачем здесь `new ArrayList<>`?
Можно же просто `return subjectStudents.getOrDefault(subject, new ArrayList<>())`

elka1907 replied:
Done

SnurnicynKonstantin reviewed (commented):
Файлы с предыдущих заданий можно не удалять. Но это не принципиально

elka1907 committed:
1. add Collections.emptyList().
2. adjusted new ArrayList<> to return subjectStudents.getOrDefault(subject, emptyList());

elka1907 committed:
1. add Collections.emptyList().
2. adjusted new ArrayList<> to return subjectStudents.getOrDefault(subject, emptyList());
3. add ;
